sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -1859,6 +1864,60 @@ public void mergePerson(String leadPersonUuid, String otherPersonUuid, boolean m
 		service.ensurePersisted(leadPerson);
 	}
 
+	private void processPersonAddressMerge(PersonDto leadPersonDto, PersonDto otherPersonDto) {
+		if (differentLocation(leadPersonDto.getAddress(), otherPersonDto.getAddress())) {
+			LocationDto otherAddress = otherPersonDto.getAddress();
+			otherAddress.setAddressType(PersonAddressType.OTHER_ADDRESS);
+			otherAddress.setAddressTypeDetails(I18nProperties.getString(Strings.messagePersonMergedAddressDescription));
+			leadPersonDto.addAddress(otherPersonDto.getAddress());
+		} else {
+			DtoHelper.copyDtoValues(leadPersonDto.getAddress(), otherPersonDto.getAddress(), false);
+		}
+	}
+
+	private boolean differentLocation(LocationDto firstAddress, LocationDto secondAddress) {
```

Comment:
Won't this method inevitably throw a null pointer if the second address does not have a country, region, district, or community because we call `getUuid()` on it without a null check?

Also, won't this method return true (i.e. different addresses) if e.g. the first address has a city, but the second has none even though that shouldn't lead to the addresses being considered different?

Reply:
rewrote the logic for determining the duplicate address